scripts/src/main/resources/scripts/functions

```diff
@@ -1257,21 +1306,77 @@ function doInstall() {
   fi
 
   # macos workaround
-  if [ ! -d "${target_path}/bin" ] && doIsMacOs && [ -d "${target_path}/Contents/Home/bin" ]
+  if [ ! -d "${target_path}/bin" ] && doIsMacOs
   then
-    doEcho "Creating symlink as workaround for ${software} on MacOS"
-    local target_software_macos=${target_path%\/*}/macos
-    if [ ! -d "${target_software_macos}" ]
+    local contents="Contents"
+    if ! [ -d "${target_path}/${contents}" ]
+    then
+      pushd "${target_path}" > /dev/null || exit 255
```

Comment:
I tried the symlink functionality on the M1 and get the following error message: 

```
m1@3dc6a48e-5e10-4e97-a481-1c8483f2748d test_doInstall % devon java setup
Success: run command mkdir
Starting installation of java in version 19_36 to /Users/m1/Documents/Alex_Dev/test_doInstall/software/java
Trying to download java-19_36-mac.tar.gz from https://github.com/adoptium/temurin19-binaries/releases/download/jdk-19%2B36/OpenJDK19U-jdk_aarch64_mac_hotspot_19_36.tar.gz
Artifact already exists at /Users/m1/Downloads/devonfw-ide/java-19_36-mac.tar.gz
To force update please delete the file and run again.
Success: run command tar
Successfully extracted archive java-19_36-mac.tar.gz to /Users/m1/Documents/Alex_Dev/test_doInstall/updates/extracted/java
Success: run command mv
Successfully installed java
Linking ../symlinktest/default/java/19_36 to /Users/m1/Documents/Alex_Dev/test_doInstall/software/java
/Users/m1/Documents/Alex_Dev/test_doInstall/scripts/command/../functions: line 1314: pushd: /Users/m1/Documents/Alex_Dev/test_doInstall/software/java: No such file or directory
```

Reply:
I updated the documentation and hope that my guess was correct. Therefore I resolve this conversation.
If my guess was incorrect, please file a new bug issue for this problem that is IMHO not related to this PR.